### DISCUSSION TITLE
How to set static-web-server to serve file as plain text instead of binary?

### DISCUSSION THREAD

**Discussion by kincsescsaba** [static-web-server/static-web-server#476] [Q&A] [answered]
If I copy a `bpmn` file (it's an XML-based markup text) under `/public`, apparently, I get binary content in my browser. The content is returned with the type `application/octet-stream`. Can I provide a configuration to map this specific extension to be served as `text/plain` or anything that won't mess up the character stream?

**Comment by joseluisq** [comment, marked as answer]
> If I copy a bpmn file (it's an XML-based markup text) under /public, apparently, I get binary content in my browser. The content is returned with the type application/octet-stream.

It is probably expected because the library used by SWS cannot guess that extension so it can map its content type. In that situation, SWS falls back to `application/octet-stream` as a "safe-bed".

> Can I provide a configuration to map this specific extension to be served as text/plain or anything that won't mess up the character stream?

Yes, you can by using [Custom HTTP Headers](https://static-web-server.net/features/custom-http-headers/) via a config file. Like this example.

**my-config.toml**

```toml
[general]
root = "/public"

[advanced]

[[advanced.headers]]
source = "**/*.bpmn"
headers = { Content-Type = "text/plain" }
```

And run SWS like `static-web-server -w /my-config.toml`
